WordPress/src/main/java/org/wordpress/android/ui/posts/PostUtils.java

```diff
@@ -283,4 +284,23 @@ public static int indexOfPostInList(final PostModel post, final List<PostModel>
         }
         return list;
     }
+
+    static boolean shouldPublishImmediately(PostModel postModel) {
+        boolean optionAvailable = shouldPublishImmediatelyOptionBeAvailable(postModel);
+        Date pubDate = DateTimeUtils.dateFromIso8601(postModel.getDateCreated());
+        Date now = new Date();
+        // For drafts with publish dates in the past, we should publish immediately
+        return optionAvailable && !pubDate.after(now);
```

Comment:
We could avoid creating the two `Date` objects by returning `false` immediately if `!shouldPublishImmediatelyOptionBeAvailable(postModel)`